sklearn/random_projection.py

```diff
@@ -437,6 +530,27 @@ def _n_features_out(self):
         """
         return self.n_components
 
+    @available_if(lambda self: self.fit_inverse_transform)
+    def inverse_transform(self, X):
+        """Project data back to its original space.
```

Comment:
I don't understand why only exposing `inverse_transform` if the inverse were computed during fit. We could just compute the inverse components if necessary from the `components_` attribute.

Reply:
That's what I initially did, if I recall correctly, but I was asked to do this instead, I think it was because that's the pattern used in other classes. I'm not sure what the original rationale was, but perhaps it's because some people would be surprised if the method is available but super slow upon the first call only. It seems reasonable to hide it, especially since it's probably not going to be called very often, so if you really want it, just set `fit_inverse_transform=True`.

Reply:
I don't mean to always do the computation of the inverse components in `inverse_transform`. What I proposed was to keep the `fit_inverse_transform` (not sure about the name btw) parameter and always expose `inverse_transform`. If the param is set to True, the inverse components are computed during fit and stored. If it's set to false they are computed during inverse_transform and not stored.

This parameter only exists in KernelPCA but I'm not sure the use case is really the same. We don't necessarily have to follow the same pattern.

Let's see what others think about that.

Reply:
After an irl discussion with @ogrisel, we agreed on always exposing the `inverse_transformed` method.